src/extensions/default/HealthData/HealthDataPreview.js

```diff
@@ -42,36 +42,36 @@ define(function (require, exports, module) {
      * Show the dialog for previewing the Health Data that will be sent.
      */
     function previewHealthData() {
-        var result = new $.Deferred();
+        var result = new $.Deferred(),
+            content;
 
         HealthDataManager.getHealthData().done(function (healthDataObject) {
-            var combinedHealthAnalyticsData = HealthDataManager.getAnalyticsData(),
-                content;
-            combinedHealthAnalyticsData = [healthDataObject, combinedHealthAnalyticsData ];
-            content = JSON.stringify(combinedHealthAnalyticsData, null, 4);
-            content = _.escape(content);
-            content = content.replace(/ /g, "&nbsp;");
-            content = content.replace(/(?:\r\n|\r|\n)/g, "<br />");
+            HealthDataManager.getAnalyticsData().done(function (ingestData) {
 
-            var hdPref   = prefs.get("healthDataTracking"),
-                template = Mustache.render(HealthDataPreviewDialog, {Strings: Strings, content: content, hdPref: hdPref}),
-                $template = $(template);
+                var combinedHealthAnalyticsData = [healthDataObject, ingestData ];
+                content = JSON.stringify(combinedHealthAnalyticsData, null, 4);
+                content = _.escape(content);
+                content = content.replace(/ /g, "&nbsp;");
+                content = content.replace(/(?:\r\n|\r|\n)/g, "<br />");
 
-            Dialogs.addLinkTooltips($template);
+                var hdPref   = prefs.get("healthDataTracking"),
+                    template = Mustache.render(HealthDataPreviewDialog, {Strings: Strings, content: content, hdPref: hdPref}),
+                    $template = $(template);
 
-            Dialogs.showModalDialogUsingTemplate($template).done(function (id) {
+                Dialogs.addLinkTooltips($template);
 
-                if (id === "save") {
-                    var newHDPref = $template.find("[data-target]:checkbox").is(":checked");
-                    if (hdPref !== newHDPref) {
-                        prefs.set("healthDataTracking", newHDPref);
+                Dialogs.showModalDialogUsingTemplate($template).done(function (id) {
+
+                    if (id === "save") {
+                        var newHDPref = $template.find("[data-target]:checkbox").is(":checked");
+                        if (hdPref !== newHDPref) {
+                            prefs.set("healthDataTracking", newHDPref);
+                        }
                     }
-                }
+                });
+                return result.resolve();
```

Comment:
what is this dialog for ? Should the result be resolved once the dialog is shown ?

Reply:
dialog is to display the healthData json and adobe analytics json.
result should be resolved when dialog is displayed.

Reply:
then should we not move it inside the function ?

Reply:
or maybe we can add an always() and resolve it there.